pandas/tests/groupby/test_categorical.py

```diff
@@ -337,14 +338,18 @@ def test_apply(ordered):
     tm.assert_series_equal(result, expected)
 
 
-def test_observed(observed):
+def test_observed(observed, using_infer_string, request):
     # multiple groupers, don't re-expand the output space
     # of the grouper
     # gh-14942 (implement)
     # gh-10132 (back-compat)
     # gh-8138 (back-compat)
     # gh-8869
 
+    if not observed and using_infer_string:
+        mark = pytest.mark.xfail(reason="fill_value=0 invalid for string dtype")
```

Comment:
Should this test be fixed in the future? If so, okay to xfail - otherwise I'd prefer to test for the exception with `pytest.raises`. xfails have a perf impact on running the tests.